### ISSUE TITLE
Fix keyboard navigation for navigation menu in mobile web

### ISSUE THREAD

**Issue by julianguyen** [ifmeorg/ifme#2159]
# Description

There are two issues with the keyboard navigation for the navigation menu in mobile web:

1. When a user uses the `Tab` key to navigate through a page, the navigation menu is opened. It should only be opened after the `Tab` key focuses on the hamburger icon and the user hits the `Enter` key to open it.
2. There is no focus trap on the navigation menu when it's opened (as a modal)

The following recent [PR for "[FIX] Keyboard navigation for modals"](https://github.com/ifmeorg/ifme/pull/2127) by @maniSHarma7575 fixed the issue in other areas. So you'll want to dig deeper into this part of the codebase as to why it's not working for the navigation menu in mobile web.

This navigation menu is part of the [Header component](http://design.if-me.org/?path=/docs/components-header--no-active-link).

![Oct-08-2022 08-12-57](https://user-images.githubusercontent.com/3010728/194714471-216410f3-ab78-46cf-9421-43b3bd0054b6.gif)

---

Please assign yourself (via the Assignees dropdown), if you do want to work on this issue. Can't find yourself? You need to [join our organization](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

Check out our [Picking Up Issues](https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues) guide if you haven't already!
